### PR TITLE
Implemented .net core version

### PR DESCRIPTION
**Pull request by meanin**
Support for:
- health checks
- readiness checks

Implemented:
- graceful shutdown, ~~(not working at this time, issue [here](https://github.com/dotnet/coreclr/issues/20765))~~

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,339 @@ demo/.idea/
 demo/demo
 demo/demo_linux
 demo/vendor/
+
+
+## Ignore Visual Studio temporary files, build results, and
```
Comment (wojciech12) [inline .gitignore:9]: Please add prefix demo_net to the .gitignore rules for the C# implementation

File: demo_net/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.dockerignore
+.env
+.git
+.gitignore
+.vs
+.vscode
+*/bin
+*/obj
+**/.toolstarget
```
Comment (wojciech12) [inline demo_net/.dockerignore]: newline :)

**Review by wojciech12 — CHANGES_REQUESTED** [review]

File: demo_net/ZeroDowntimeDeployment/Middlewares/GracefulShutdownMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
+﻿using System.Runtime.Loader;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace ZeroDowntimeDeployment.Middlewares
+{
+    public class GracefulShutdownMiddleware
```
Comment (wojciech12) [inline demo_net/ZeroDowntimeDeployment/Middlewares/GracefulShutdownMiddleware.cs:10]: Could we add more logging and add 5 seconds sleep before we do the graceful shutdown. With 5 seconds sleep, we can test whether the code below works.

File: demo_net/ZeroDowntimeDeployment/Middlewares/LivenessProbeMiddleware.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            var path = context.Request.Path.Value.ToLowerInvariant();
+            switch (path)
+            {
+                case "/probe":
+                // Support for Zero downtime deployment with kubernetes scenarios
+                case "/healthz":
```
Comment (wojciech12) [inline demo_net/ZeroDowntimeDeployment/Middlewares/LivenessProbeMiddleware.cs:35]: Please also return: OK <hostname> and NOT OK <hostname>.
So, during the demo, it is clear which instances of our demo are unhealthy.

**Comment by meanin** [comment]
Solution tested on docker interactive. Graceful Shutdown Middleware works as expected. 

Thanks for this journey :)

**Comment by wojciech12** [comment]
@meanin  Thank you for your contribution. I will merge it.

**Pull request merged**